### PR TITLE
feat: [CDS-74369]: add aborted by information in execution listing pages

### PR DESCRIPTION
**Pull request by PratyushGar**
### Summary

- Added abortedBy user and timestamp information in execution listing details
- Updated the TimeAgoPopover UTC/Local UX to be inline and concise as per new UX

| Before | After |
| -- | -- |
| image | image |

**CD module:**

image

**Other Modules:**

image

image
image
image
image

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by vikrant-harness — APPROVED** [review]

File: src/modules/10-common/components/TimeAgoPopover/TimeAgoPopover.tsx [hunk 1]
```diff
@@ -31,24 +59,7 @@ export const TimeAgoPopover: React.FC<TimeAgoPopoverProps> = props => {
...
-        <Layout.Horizontal margin={{ top: 'small' }} flex={{ justifyContent: 'flex-start' }} spacing={'small'}>
-          <Text font={{ size: 'xsmall', weight: 'semi-bold' }} color={Color.GREY_200}>
-            TIME
-          </Text>
-          <Text font={{ size: 'small', weight: 'bold' }} color={Color.WHITE}>
-            {new Date(time).toLocaleTimeString()}
-          </Text>
-        </Layout.Horizontal>
-      </Container>
+      {TimeAgoContent(time, 'medium')}
```
Comment (ishantmehtaharness) [inline src/modules/10-common/components/TimeAgoPopover/TimeAgoPopover.tsx:62]: We should use `TimeAgoContent` as a react component, and pass it props
<TimeAgoContent time={time} padding='medium'/>

File: src/modules/10-common/components/TimeAgoPopover/TimeAgoPopover.tsx [hunk 2]
```diff
@@ -7,17 +7,45 @@
...
 
 export interface TimeAgoPopoverProps extends TextProps, React.ComponentProps<typeof ReactTimeago> {
   time: number
   popoverProps?: IPopoverProps
   icon?: IconName
   className?: string
 }
+
+export function TimeAgoContent(time: number, padding?: StyledProps['padding']): JSX.Element {
+  const { getString } = useStrings()
```
Comment (rameshx) [inline src/modules/10-common/components/TimeAgoPopover/TimeAgoPopover.tsx:25]: `TimeAgoContent` should be a function component as it's using hooks

File: src/modules/10-common/components/TimeAgoPopover/TimeAgoPopover.tsx (hunk 1 above)
Comment (rameshx) [inline src/modules/10-common/components/TimeAgoPopover/TimeAgoPopover.tsx:62]: time and 'medium' can be passed as props

**Review by ishantmehtaharness — COMMENTED** [review]
added comments

File: src/modules/70-pipeline/components/ExecutionCard/TimePopoverWithLocal.tsx [hunk 3]
```diff
@@ -20,37 +20,44 @@ export interface TimePopoverProps extends TextProps, React.ComponentProps<typeof
...
 }
+export const DATE_PARSE_FORMAT = 'MMM DD, YYYY hh:mm:ss A'
 
 enum TimeZone {
   UTC = 'UTC',
   LOCAL = 'LOCAL'
 }
 
-function TimeDate(timeZone: string, time: number): JSX.Element {
+export function TimeDate(time: number): JSX.Element {
```
Comment (rameshx) [inline src/modules/70-pipeline/components/ExecutionCard/TimePopoverWithLocal.tsx:30]: let's make this a function component too

**Review by rameshx — CHANGES_REQUESTED** [review]

File: src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx [hunk 4]
```diff
@@ -249,17 +251,50 @@ export const StatusCell: CellType = ({ row }) => {
...
+                {userName}
+              </Text>
+              <Text color={Color.PRIMARY_1} font={{ variation: FontVariation.SMALL_BOLD }}>
+                {email}
+              </Text>
+            </Layout.Vertical>
+          </Layout.Horizontal>
+          <Layout.Horizontal>
+            <Icon name="timer" size={15} color={Color.GREY_100} padding={{ right: 'small', top: 'small' }} />
+            {module === 'cd' ? TimeDate(createdAt) : TimeAgoContent(createdAt)}
```
[the remark this answers is not in the document]
Comment (PratyushGar) [reply src/modules/70-pipeline/pages/execution-list/ExecutionListTable/ExecutionListCells.tsx]: Renamed the components for better understanding. The primary difference is that CD popover shows UTC and local time.

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good, thanks for addressing the comments

**Review by rameshx — APPROVED** [review]

**Review by bhavya-sinha — APPROVED** [review]

**Pull request merged**